src/SFA.DAS.AANHub.Api/Controllers/ApprenticesController.cs

```diff
@@ -30,39 +31,56 @@ public ApprenticesController(ILogger<ApprenticesController> logger, IMediator me
         [Produces("application/json")]
         [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
         [ProducesResponseType(typeof(string), StatusCodes.Status201Created)]
-        public async Task<IActionResult> CreateApprentice([FromHeader(Name = Constants.PostRequestHeaders.RequestedByUserHeader), Required] Guid? userId, CreateApprenticeModel request)
+        public async Task<IActionResult> CreateApprentice([FromHeader(Name = Constants.PostRequestHeaders.RequestedByUserHeader)] [Required] Guid? userId,
+            CreateApprenticeModel request)
         {
-            _logger.LogInformation("AAN Hub API: Received command to add apprentice by ApprenticeId: {apprenticeId} and UserId: {userId}", request.ApprenticeId, userId);
+            _logger.LogInformation("AAN Hub API: Received command to add apprentice by ApprenticeId: {apprenticeId} and UserId: {userId}",
+                request.ApprenticeId,
+                userId);
 
             CreateApprenticeMemberCommand command = request;
             command.RequestedByMemberId = userId;
 
             var response = await _mediator.Send(command);
-            return new CreatedAtActionResult(nameof(ApprenticesController.GetApprentice), "Apprentices", new { apprenticeid = request.ApprenticeId }, response);
+
+            return GetPostResponse(response,
+                new BaseRequestDetails
+                {
+                    ActionName = nameof(GetApprentice),
```

Comment:
We can avoid this in get methods.

Reply:
Get logs have been removed but I've left this as is because we need the ActionName, and Controller name to build the response header for the GET request:

`location: https://localhost:7299/api/Apprentices/12341111`

I wasn't aware we were doing this, but Ravi has been implementing them into the various POST endpoints he's worked on, based on feedback from yourself?

Let me know if not.

Reply:
I put the comment on the wrong endpoint. I meant to put this on GET endpoint not on POST